build-conventions/src/main/java/org/elasticsearch/gradle/internal/conventions/LicensingPlugin.java

```diff
@@ -34,24 +33,23 @@ public LicensingPlugin(ProviderFactory providerFactory) {
     public void apply(Project project) {
         Provider<String> revision = project.getRootProject().getPlugins().apply(GitInfoPlugin.class).getRevision();
         Provider<String> licenseCommitProvider = providerFactory.provider(() ->
-             isSnapshotVersion(project) ? revision.get() : "v" + project.getVersion().toString()
+             isSnapshotVersion(project) ? revision.get() : "v" + project.getVersion()
         );
 
-        MapProperty<String, String> licensesProperty = project.getObjects().mapProperty(String.class, String.class);
         Provider<String> projectLicenseURL = licenseCommitProvider.map(licenseCommit -> ELASTIC_LICENSE_URL_PREFIX +
                 licenseCommit + ELASTIC_LICENSE_URL_POSTFIX);
         // But stick the Elastic license url in project.ext so we can get it if we need to switch to it
         project.getExtensions().getExtraProperties().set("elasticLicenseUrl", projectLicenseURL);
 
-        MapProperty<String, String> convention = licensesProperty.convention(
+        MapProperty<String, String> licensesProperty = project.getObjects().mapProperty(String.class, String.class).convention(
                 providerFactory.provider((Callable<Map<? extends String, ? extends String>>) () -> Map.of(
                         "Server Side Public License, v 1", "https://www.mongodb.com/licensing/server-side-public-license",
                         "Elastic License 2.0", projectLicenseURL.get())
                 )
         );
+
         // Default to the SSPL+Elastic dual license
-        project.getExtensions().getExtraProperties().set("licenseCommit", licenseCommitProvider);
-        project.getExtensions().getExtraProperties().set("projectLicenses", convention);
+        project.getExtensions().getExtraProperties().set("projectLicenses", licensesProperty);
     }
 
     private boolean isSnapshotVersion(Project project) {
```

Comment:
Can we ditch this and just use `BuildParams.isSnapshot`?

Reply:
BuildParams is build-tools-internal, this is build-conventions. so unfortunately we cannot